nixos/modules/services/misc/sourcehut/default.nix

```
@@ -77,6 +77,14 @@ let
       type = types.path;
       apply = s: "<" + toString s;
     };
+    api-origin = mkOption {
+      description = lib.mdDoc "Origin URL for API, 100 more than web.";
+      type = types.str;
+      default = "http://${cfg.listenAddress}:${toString (cfg.${srv}.port + 100)}";
+      defaultText = lib.literalMD ''
+        `"http://''${`[](#opt-services.sourcehut.listenAddress)`}:''${toString (`[](#opt-services.sourcehut.${srv}.port)` + 100)}"`
+      '';
+    };
```

Let's define `opt = options.services.sourcehut;` alongside `cfg`. Not only is it a bit shorter - it also ensures the option actually exists.

```suggestion
      defaultText = lib.literalExpression ''
        "http://''${config.${opt.listenAddress}}:''${toString (config.${opt.${srv}.port} + 100)}"
      '';
    };
```

---

Again, I'd rather do this as a separate PR. It's not breaking anything currently, so let's keep at this for this PR.